### PR TITLE
Update pca.cpp

### PR DESCRIPTION
**Pull request by sturkmen72**
see http://answers.opencv.org/question/85521

### CODE REVIEW

**Comment by Dikay900** [comment]
A bit explanation for the devs so this can be merged: The fs variable is a bit confusing for the user since the correct class used there is FileNode not FileStorage
Can you explain why the empty assert is being thrown anyway there? What has to be filled in a filenode to get through this assert?

**Comment by mshabunin** [comment]
`empty` check should be retained.
PCA loading should be done by calling: `pca.read(fs.root())`

**Comment by sturkmen72** [comment]
@mshabunin empty check retained.

File: modules/core/include/opencv2/core.hpp [hunk 1]
```diff
@@ -2316,11 +2316,11 @@ class CV_EXPORTS PCA
     The operator performs %PCA of the supplied dataset. It is safe to reuse
     the same PCA structure for multiple datasets. That is, if the structure
     has been previously used with another dataset, the existing internal
-    data is reclaimed and the new eigenvalues, @ref eigenvectors , and @ref
+    data is reclaimed and the new @ref eigenvalues, @ref eigenvectors , and @ref
```
Comment (mshabunin) [inline modules/core/include/opencv2/core.hpp]: Comma after word `@ref eigenvectors` can be removed. Same for _read_ and _write_ documentation.

File: modules/core/src/pca.cpp [hunk 2]
```diff
@@ -158,15 +158,14 @@ void PCA::write(FileStorage& fs ) const
...
 }
 
-void PCA::read(const FileNode& fs)
+void PCA::read(const FileNode& fn)
 {
-    CV_Assert( !fs.empty() );
-    String name = (String)fs["name"];
-    CV_Assert( name == "PCA" );
+    CV_Assert( !fn.empty() );
+    CV_Assert( String(fn["name"]) == "PCA" );
```
Comment (mshabunin) [inline modules/core/src/pca.cpp]: It is better to explicitly cast to _String_ to avoid ambiguity: `(String)fn["name"] == "PCA"`

**Comment by mshabunin** [comment]
@sturkmen72 , thank you! I have only two comments left.

**Comment by sturkmen72** [comment]
@mshabunin Done!

**Comment by sturkmen72** [comment]
@mshabunin your two comments are done.

**Comment by mshabunin** [comment]
:+1:

**Pull request merged**